src/test/java/org/jabref/gui/externalfiles/AutoSetFileLinksUtilTest.java

```diff
@@ -30,24 +30,33 @@
     private final BibDatabaseContext databaseContext = mock(BibDatabaseContext.class);
     private final ExternalFileTypes externalFileTypes = mock(ExternalFileTypes.class);
     private final BibEntry entry = new BibEntry(StandardEntryType.Article);
+    private Path path = null;
 
     @BeforeEach
     public void setUp(@TempDir Path folder) throws Exception {
-        Path path = folder.resolve("CiteKey.pdf");
+        path = folder.resolve("CiteKey.pdf");
         Files.createFile(path);
         entry.setCiteKey("CiteKey");
-        when(databaseContext.getFileDirectoriesAsPaths(any())).thenReturn(Collections.singletonList(path.getParent()));
-        when(externalFileTypes.getExternalFileTypeSelection()).thenReturn(new TreeSet<>(ExternalFileTypes.getDefaultExternalFileTypes()));
     }
 
     @Test
-    public void test() throws Exception {
+    public void testFindAssociatedNotLinkedFilesSuccess() throws Exception {
         // Due to mocking the externalFileType class, the file extension will not be found
 
+        when(databaseContext.getFileDirectoriesAsPaths(any())).thenReturn(Collections.singletonList(path.getParent()));
```

Comment:
Why did you move the mockings here?

Reply:
This mocking of getFileDirectoriesAsPaths method returns non empty values, which is not applicable for the newly added test case testFindAssociatedNotLinkedFilesForEmptySearchDir. The new test case expects that getFileDirectoriesAsPaths would return empty list.

Reply:
Okay, but the externalFileType should be independent of this

Reply:
yeah, agreed. I will move that one back to setup.